### ISSUE TITLE
Document relationship between overlapping `*-no-unknown` rules

### ISSUE THREAD

**Issue by HHaoWang** [stylelint/stylelint#9037]
### What minimal example or steps are needed to reproduce the bug?

```
/* test.css */
.container {
  width: 100rpx;
  height: 50rpx;
}
```

### What minimal configuration is needed to reproduce the bug?

```
{
  "rules": {
    "unit-no-unknown": [
      true,
      {
        "ignoreUnits": ["rpx"]
      }
    ],
    "declaration-property-value-no-unknown": true
  }
}
```

### How did you run Stylelint?

```
npx stylelint "test.css"
```

### Which Stylelint-related dependencies are you using?

```
{
  "stylelint": "17.1.1"
}
```

### What did you expect to happen?

Since I've configured unit-no-unknown to ignore the rpx unit, I expected no errors to be reported for values like 100rpx and 50rpx.

Alternatively, I expected the declaration-property-value-no-unknown rule to respect the ignoreUnits configuration from unit-no-unknown, so that custom units allowed by unit-no-unknown would automatically be considered valid by declaration-property-value-no-unknown.

### What actually happened?

The following problems were reported:

```
test.css
 2:10  ✖  Unexpected unknown property value "100rpx"  declaration-property-value-no-unknown
 3:11  ✖  Unexpected unknown property value "50rpx"   declaration-property-value-no-unknown
```

Even though rpx is listed in the ignoreUnits of unit-no-unknown, the declaration-property-value-no-unknown rule still reports these values as unknown.

To fix this, I have to also configure declaration-property-value-no-unknown:
```
{
  "rules": {
    "unit-no-unknown": [
      true,
      {
        "ignoreUnits": ["rpx"]
      }
    ],
    "declaration-property-value-no-unknown": [
      true,
      {
        "ignoreProperties": {
          '/.+/': ['/[0-9]+\x20*rpx/'],
        }
      }
    ]
  }
}
```

### Do you have a proposal to fix the bug?

I'm not sure if this is a bug or intended behavior, but I believe there's room for improvement in how these two rules interact.

**My preferred solution would be:**

**Configuration sharing**: Allow `declaration-property-value-no-unknown` to automatically respect the `ignoreUnits` configuration from `unit-no-unknown`. This would:
- Eliminate redundant configuration
- Provide a more intuitive developer experience
- Make the relationship between these two rules clearer

For example, if I configure:
```json
{
  "rules": {
    "unit-no-unknown": [true, { "ignoreUnits": ["rpx"] }]
  }
}
```

Then `declaration-property-value-no-unknown` should automatically treat `rpx` as a valid unit without requiring additional configuration.

**Alternative solutions** (if configuration sharing is not feasible):

1. **Documentation improvement**: Clearly document that `unit-no-unknown` and `declaration-property-value-no-unknown` work independently and both need to be configured separately for custom units.

2. **Clarify use cases**: Document when to use `unit-no-unknown` vs `declaration-property-value-no-unknown`, as it seems the latter supersedes the former when both are enabled.

**Comment by jeddy3** [comment]
@HHaoWang Thanks for the clearly written report.

We can definitely improve people's experience here by making the documentation clearer.

The `declaration-property-value-no-unknown` rule supersedes a number of our earlier `*-no-unknown` rules, such as `unit-no-unknown` and `function-no-unknown`, for most use cases, especially when linting vanilla CSS. However, we kept the earlier rules for users of language extensions, such as SCSS or Less, because they were more compatible.

The new breed of `*-no-unknown` rules support a shared syntax through the [`languageOptions`](https://stylelint.io/user-guide/configure/#languageoptions) configuration property, for example:

```js
{
  "languageOptions": {
    "syntax": {
      "atRules": {
        "example": {
          "comment": "Example at-rule",
          "prelude": "<custom-ident>",
          "descriptors": {
            "foo": "<number>",
            "bar": "<color>"
          }
        }
      },
      "cssWideKeywords": ["my-global-value"],
      "properties": { "top": "| <--foo()>" },
      "types": { "--foo()": "--foo( <length-percentage> )" }
    }
  },
  "rules": {
    "at-rule-descriptor-no-unknown": true,
    "at-rule-descriptor-value-no-unknown": true,
    "at-rule-prelude-no-invalid": true,
    "declaration-property-value-no-unknown": true
  }
}
```

However, I just realised we don't support extending the units. I'll open an issue for that.

We [recommend extending our shared config](https://stylelint.io/user-guide/get-started), which only turns on the new breed of `*-no-unknown` rules. So to support `rpx` units right now, we suggest:

```js
/** @type {import('stylelint').Config} */
export default {
  extends: ["stylelint-config-standard"], 
  rules: {
    "declaration-property-value-no-unknown": [
      true,
      {
        "ignoreProperties": {
          '/.+/': ['/[0-9]+\x20*rpx/'],
        }
      }
    ]
  }
};
```

If we add support for extending units, that'll become:

```js
/** @type {import('stylelint').Config} */
export default {
  extends: ["stylelint-config-standard"], 
  languageOptions: {
    syntax: {
      units: { length: ["rpx"] }
    }
  },
};
```

And will be more capable. For example, it'd still detect:

```css
a { color: 10rpx; }
```

Whereas using `ignoreProperties` won't.

**Comment by jeddy3** [comment]
@HHaoWang today's [`17.6.0`](https://github.com/stylelint/stylelint/releases/tag/17.6.0) release adds supports for extending units in `languageOptions`, so you can now do:

```js
/** @type {import('stylelint').Config} */
export default {
  extends: ["stylelint-config-standard"], 
  languageOptions: {
    syntax: {
      units: { length: ["rpx"] }
    }
  },
};
```

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdilAXCADvCgIwAMATgQB4DcWAviADQgBmAlgDYwBySAWziIY1QQR4A6MAGcZLcBEycA5ghAB6AFRaUAAVwBPAjDQcBBCBVwAKAOQyjPLh0y47ASkkBhJaoYoWhpYopbWKLBsSACuXLhoWDiiuDCYUDLEANroII6Gzq64ALSQyhwqRY5IaUgUUDkAuswoiShc1SrRSCowAPIEuBxKGQnYOCgyhm5I1MQYY+Mo0ZgcuCPAbakquAAWWTlU1I0oDK04p2MMzIy0CqWqAGJWAki46nkFbtJ+5ZICAFbyViwAjyRDzHA5D4wFxuHLEHLtFKOHLXMZQpwwwolH4VKo1OrwlogJEwFEgRhYEAMIA).

**Comment by HHaoWang** [comment]
> [@HHaoWang](https://github.com/HHaoWang) today's [`17.6.0`](https://github.com/stylelint/stylelint/releases/tag/17.6.0) release adds supports for extending units in `languageOptions`, so you can now do:
> 
> ```
> /** @type {import('stylelint').Config} */
> export default {
>   extends: ["stylelint-config-standard"], 
>   languageOptions: {
>     syntax: {
>       units: { length: ["rpx"] }
>     }
>   },
> };
> ```
> [Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdilAXCADvCgIwAMATgQB4DcWAviADQgBmAlgDYwBySAWziIY1QQR4A6MAGcZLcBEycA5ghAB6AFRaUAAVwBPAjDQcBBCBVwAKAOQyjPLh0y47ASkkBhJaoYoWhpYopbWKLBsSACuXLhoWDiiuDCYUDLEANroII6Gzq64ALSQyhwqRY5IaUgUUDkAuswoiShc1SrRSCowAPIEuBxKGQnYOCgyhm5I1MQYY+Mo0ZgcuCPAbakquAAWWTlU1I0oDK04p2MMzIy0CqWqAGJWAki46nkFbtJ+5ZICAFbyViwAjyRDzHA5D4wFxuHLEHLtFKOHLXMZQpwwwolH4VKo1OrwlogJEwFEgRhYEAMIA).

It works perfectly now! I really appreciate your work. Thank you so much!
